src/box/tuple_format.c

```diff
@@ -1024,6 +1024,79 @@ tuple_format_required_fields_validate(struct tuple_format *format,
 				      void *required_fields,
 				      uint32_t required_fields_sz);
 
+/**
+ * Check if integer value in `mp_data' is within an allowed range for the given
+ * `field' type.
+ * Return -1 and raise an error if it is out of range, or 0 if all is ok.
+ */
+static int
+tuple_field_check_int_value_range(struct tuple_format *format,
+				  struct tuple_field *field,
+				  const char *mp_data)
+{
+	assert(field->type < field_type_MAX);
+	enum mp_type mp_type = mp_typeof(*mp_data);
+	assert((size_t)mp_type < CHAR_BIT * sizeof(*field_mp_type));
+
+	if (mp_type != MP_INT && mp_type != MP_UINT)
+		return 0;
+	const char *min_str, *max_str, *val_str;
+
+	switch (field->type) {
+	case FIELD_TYPE_INT8:
+	case FIELD_TYPE_INT16:
+	case FIELD_TYPE_INT32:
+	case FIELD_TYPE_INT64: {
+		int64_t min = field_type_min_value[field->type];
+		int64_t max = field_type_max_value[field->type];
+		if (mp_type == MP_INT) {
```

Comment:
I don't quite like this switch-case and the min/max constants defined for *all* types. The code would be more generalized if we used callbacks for checking values. However, it could affect performance. Need to discuss with @alyapunov.

Reply:
Discuss f2f and decided not to use callbacks.